clients/mobile/src/donationWorkflow/createUpdateDonation/useBloodRequest.ts

```diff
@@ -0,0 +1,179 @@
+    if (name === DONATION_DATE_TIME_INPUT_NAME) {
+      onDateChange(value)
+      return
+    }
+    setBloodRequestData(prevState => ({
+      ...prevState,
+      [name]: value
+    }))
+    if (name in validationRules) {
+      handleInputValidation(name as keyof BloodRequestDataErrors, value)
+    }
+  }
+
+  const handleInputValidation = (name: keyof BloodRequestDataErrors, value: string): void => {
+    const errorMsg = validateInput(value, validationRules[name])
+    setErrors(prevErrors => ({
+      ...prevErrors,
+      [name]: errorMsg
+    }))
+  }
+
+  const isButtonDisabled = useMemo(() => {
+    const hasErrors = !Object.values(errors).every(error => error === null)
+
+    const requiredFieldsFilled = Object.keys(validationRules).every((key: string) => {
+      const value = bloodRequestData[key as CredentialKeys]
+      if (typeof value === 'string') {
+        return value.trim() !== ''
+      } else if (value instanceof Date) {
+        return !isNaN(value.getTime())
+      }
+      return false
+    })
+
+    return hasErrors || !requiredFieldsFilled
+  }, [errors, bloodRequestData])
+
+  const removeEmptyAndNullProperty = (object: Record<string, unknown>): Record<string, unknown> => {
+    return Object.fromEntries(Object.entries(object).filter(([_, v]) => v != null && v !== ''))
+  }
+
+  const createBloodDonationRequest = async(): Promise<DonationResponse> => {
+    const { bloodQuantity, ...rest } = bloodRequestData
+    const locationService = new LocationService()
+    const coordinates = await locationService.getCoordinates(rest.location)
+    const finalData = {
+      ...removeEmptyAndNullProperty(rest),
+      contactNumber: formatPhoneNumber(rest.contactNumber),
+      bloodQuantity: +bloodQuantity.replace(/\b(\d+) (Bag|Bags)\b/, '$1'),
+      donationDateTime: typeof rest.donationDateTime === 'string'
+        ? new Date(rest.donationDateTime).toISOString()
+        : rest.donationDateTime.toISOString(),
+      latitude: +coordinates.lat,
+      longitude: +coordinates.lon
+    }
+    return await createDonation(finalData, fetchClient)
+  }
+
+  const updateBloodDonationRequest = async(): Promise<DonationResponse> => {
+    const { bloodQuantity, location, neededBloodGroup, ...rest } = bloodRequestData
+    const finalData = {
+      ...removeEmptyAndNullProperty(rest),
+      contactNumber: formatPhoneNumber(rest.contactNumber),
+      bloodQuantity: +bloodQuantity.replace(/\b(\d+) (Bag|Bags)\b/, '$1'),
+      donationDateTime: new Date(rest.donationDateTime).toISOString()
+    }
+    return await updateDonation(finalData, fetchClient)
+  }
+
+  const handlePostNow = async(): Promise<void> => {
+    try {
+      setLoading(true)
+      const validateDonationDate = validateDonationDateTime(new Date(bloodRequestData.donationDateTime).toISOString())
+      if (validateDonationDate !== null) {
+        setErrorMessage(validateDonationDate)
+        return
+      }
+      let response: DonationResponse
```

Comment:
i feel like you can make this better. Would you like to try to not use let

Reply:
Yes vai,

```
const response: DonationResponse = isUpdating
        ? await updateBloodDonationRequest()
        : await createBloodDonationRequest()
```

Reply:
sorted